src/ddmd/traits.d

```diff
@@ -370,6 +373,73 @@ extern (C++) Expression pointerBitmap(TraitsExp e)
     return ale;
 }
 
+extern (C++) Expression externDmangle(TraitsExp e, Scope* sc)
```

Comment:
> So it might be quite easy to do the replacements in druntime.

I'm afraid it isn't that easy. You'll have to reencode the type in case it contains identical identifiers or symbols also part of the qualified name. All back references might change due to this.

I just realized that the current implementation of externDmangle also fails for symbols made of multiple identifiers in case these are also used in the type (e.g. a package name). Creating new symbols in this case is wrong, they must be resolved. Alternatively the mangler should not compare symbols by pointer, but their string representation. This might cost some performance, though.